server/tests/batch_comparison/test_multi_contest_batch_comparison.py

```diff
@@ -0,0 +1,1003 @@
+            for i, sheet_results in enumerate(results)
+        ],
+    )
+
+
+def test_multi_contest_batch_comparison_jurisdiction_upload_validation(
+    client: FlaskClient,
+    election_id: str,
+    jurisdiction_ids: List[str],
+    contest_ids,  # pylint: disable=unused-argument
+    election_settings,  # pylint: disable=unused-argument
+    manifests,  # pylint: disable=unused-argument
+):
+    test_cases = [
+        # Success cases
+        (jurisdiction_ids[0], io.BytesIO(VALID_BATCH_TALLIES[0]), None),
+        (jurisdiction_ids[1], io.BytesIO(VALID_BATCH_TALLIES[1]), None),
+        (jurisdiction_ids[2], io.BytesIO(VALID_BATCH_TALLIES[2]), None),
+        # Error cases
+        (
+            jurisdiction_ids[0],
+            # Missing contest 2 columns for jurisdiction with both contests
+            io.BytesIO(b"Batch Name,Candidate 1,Candidate 2\n"),
+            "Missing required columns: Candidate 3, Candidate 4.",
+        ),
+        (
+            jurisdiction_ids[0],
+            # Missing contest 1 columns for jurisdiction with both contests
+            io.BytesIO(b"Batch Name,Candidate 3,Candidate 4\n"),
+            "Missing required columns: Candidate 1, Candidate 2.",
+        ),
+        (
+            jurisdiction_ids[0],
+            # Extra column
+            io.BytesIO(
+                b"Batch Name,Candidate 1,Candidate 2,Candidate 3,Candidate 4,Candidate 5\n"
+            ),
+            "Found unexpected columns. Allowed columns: Batch Name, Candidate 1, Candidate 2, Candidate 3, Candidate 4.",
+        ),
+        (
+            jurisdiction_ids[1],
+            # Missing contest 1 column for jurisdiction with only contest 1
+            io.BytesIO(b"Batch Name,Candidate 1\n"),
+            "Missing required column: Candidate 2.",
+        ),
+        (
+            jurisdiction_ids[1],
+            # Including contest 2 columns for jurisdiction with only contest 1
+            io.BytesIO(b"Batch Name,Candidate 1,Candidate 2,Candidate 3,Candidate 4\n"),
+            "Found unexpected columns. Allowed columns: Batch Name, Candidate 1, Candidate 2.",
+        ),
+        (
+            jurisdiction_ids[0],
+            # Too many votes for contest 1
+            io.BytesIO(
+                b"Batch Name,Candidate 1,Candidate 2,Candidate 3,Candidate 4\n"
+                b"Batch 1,100,1,0,0\n"
+                b"Batch 2,0,0,0,0\n"
+                b"Batch 3,0,0,0,0\n"
+                b"Batch 4,0,0,0,0\n"
+                b"Batch 5,0,0,0,0\n"
+                b"Batch 6,0,0,0,0\n"
+                b"Batch 7,0,0,0,0\n"
+                b"Batch 8,0,0,0,0\n"
+                b"Batch 9,0,0,0,0\n"
+                b"Batch 10,0,0,0,0\n"
+            ),
+            'The total votes for batch "Batch 1" (101 votes) cannot exceed 100 - '
+            "the number of ballots from the manifest (100 ballots) "
+            "multiplied by the number of votes allowed for the contest (1 vote per ballot).",
```

Comment:
Seems like this error message should include the name of the contest in question

Reply:
👍 https://github.com/votingworks/arlo/pull/1884/commits/7422a1ef47b44b5ed8c248cd74f3f782b45bc229